server/command.go

```diff
@@ -209,6 +209,15 @@ func (p *Plugin) ExecuteCommand(_ *plugin.Context, args *model.CommandArgs) (*mo
 		return &model.CommandResponse{}, nil
 	}
 
+	if action == commandTriggerSetChannelWelcome || action == commandTriggerGetChannelWelcome || action == commandTriggerDeleteChannelWelcome {
+		hasPermissionTo := p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PERMISSION_MANAGE_SLASH_COMMANDS)
```

Comment:
@hanzei Is `model.PERMISSION_MANAGE_SLASH_COMMANDS` the right permission to use here? We want to know if the user is a channel admin. Maybe we should use `model.PermissionManageChannelRoles`?

Permissions for reference: https://github.com/mattermost/mattermost-server/blob/9dc85ea092fa753be62d89862f723afba374e291/model/permission.go#L2113

Reply:
`PermissionManageChannelRoles` sound good to me :+1:

Reply:
@mickmister  actually this makes sense. I missed this permission earlier. Will update the PR with the comment below